app/payment-link-v2/reference/get-index.controller.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+const { response } = require('../../utils/response')
+const { getSessionVariable } = require('../../utils/cookie')
+const { NotFoundError } = require('../../errors')
+const getBackLinkUrl = require('./get-back-link-url')
+
+module.exports = (req, res, next) => {
+  const product = req.product
+
+  const data = {
+    productExternalId: product.externalId,
+    productName: product.name
+  }
+
+  if (!product.reference_enabled) {
+    return next(new NotFoundError('Attempted to access reference page with a product that auto-generates references.'))
+  }
+
+  const sessionReferenceNumber = getSessionVariable(req, 'referenceNumber')
+
+  data.backLinkHref = getBackLinkUrl(sessionReferenceNumber, product)
+
+  if (sessionReferenceNumber) {
+    data.referenceNumber = sessionReferenceNumber
+  }
+
+  if (product.reference_label) {
```

Comment:
I don't think the `reference_label` can be blank - if it were this would mean we would display "Please enter your" as the input label. If anything we should throw an error if it's blank.

Reply:
The `rereference_label` - cannot be blank so will remove the `if` around it.